### PR TITLE
CT-133 - Create a NumberPresenter and NumberEditor in TemplateMaster

### PR DESCRIPTION
**Pull request by gvjoshi25**
CT-133 - Create a NumberPresenter and NumberEditor in TemplateMaster
CT-134 - Create a BigIntegerParam

### CODE REVIEW

File: library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    public static final NumberPresenter DOLLARS_NO_CENTS = new NumberPresenter("$#,##0");
+    public static final NumberPresenter DOLLARS_CENTS = new NumberPresenter("$#,##0.00");
+    public static final NumberPresenter WHOLE_NO_GROUPING = new NumberPresenter("0");
+    public static final NumberPresenter WHOLE_GROUPING = new NumberPresenter("#,##0");
+    public static final NumberPresenter DECIMAL_NO_GROUPING = new NumberPresenter("0.#");
+    public static final NumberPresenter DECIMAL_GROUPING = new NumberPresenter("#,##0.#");
+
+    private DecimalFormat decimalFormat;
+
+    public NumberPresenter(String decimalFormat) {
```
Comment (DeleTaylor) [inline library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java]: It seems we've removed the overloaded constructor that accepts a DecimalFormat instance.  Please add it back.

File: library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class NumberPresenter implements FieldValuePresenter {
+
+    public static final NumberPresenter DOLLARS_NO_CENTS = new NumberPresenter("$#,##0");
+    public static final NumberPresenter DOLLARS_CENTS = new NumberPresenter("$#,##0.00");
+    public static final NumberPresenter WHOLE_NO_GROUPING = new NumberPresenter("0");
+    public static final NumberPresenter WHOLE_GROUPING = new NumberPresenter("#,##0");
+    public static final NumberPresenter DECIMAL_NO_GROUPING = new NumberPresenter("0.#");
+    public static final NumberPresenter DECIMAL_GROUPING = new NumberPresenter("#,##0.#");
+
+    private DecimalFormat decimalFormat;
```
Comment (DeleTaylor) [inline library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java]: Let's make the field final.
```suggestion
    private final DecimalFormat decimalFormat;
```

File: library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+public class NumberEditor implements FieldValueEditor {
+
+    public static final NumberPresenter DOLLARS_NO_CENTS = new NumberPresenter("$#,##0");
+    public static final NumberPresenter DOLLARS_CENTS = new NumberPresenter("$#,##0.00");
+    public static final NumberPresenter WHOLE_NO_GROUPING = new NumberPresenter("0");
+    public static final NumberPresenter WHOLE_GROUPING = new NumberPresenter("#,##0");
+    public static final NumberPresenter DECIMAL_NO_GROUPING = new NumberPresenter("0.#");
+    public static final NumberPresenter DECIMAL_GROUPING = new NumberPresenter("#,##0.#");
+
+    private DecimalFormat decimalFormat;
```
Comment (DeleTaylor) [inline library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java]: Let's make the field final.
```suggestion
    private final DecimalFormat decimalFormat;
```

File: library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public NumberEditor(String decimalFormat) {
```
[the remark this answers is not in the document]
Comment (gvjoshi25) [reply library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java]: I thought we agreed to remove that constructor. But that's fine. I am adding this constructor for both NumberPreseneter & NumberEditor.

File: library/src/main/java/com/northconcepts/templatemaster/form/param/BigIntegerParam.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        super(valueAsString);
+    }
+
+    public BigIntegerParam(BigInteger value) {
+        super(value);
+    }
+
+    @Override
+    protected BigInteger parse(String valueAsString) throws Throwable {
+        if (valueAsString == null) {
```
[the remark this answers is not in the document]
Comment (gvjoshi25) [reply library/src/main/java/com/northconcepts/templatemaster/form/param/BigIntegerParam.java]: Yes, this check is not required. Removing it.

File: library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java (hunk 1 above)
Comment (gvjoshi25) [reply library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java]: I thought we agreed to remove that constructor. But that's fine. I am adding this constructor for both NumberPreseneter & NumberEditor.

File: library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java (hunk 2 above)
Comment (gvjoshi25) [reply library/src/main/java/com/northconcepts/templatemaster/form/presenter/NumberPresenter.java]: Done.

File: library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java (hunk 3 above)
Comment (gvjoshi25) [reply library/src/main/java/com/northconcepts/templatemaster/form/editor/NumberEditor.java]: Done.

**Pull request merged**